pynetket/python_utils.py

```diff
@@ -0,0 +1,199 @@
+#!/usr/bin/env python
+# Copyright 2018 The Simons Foundation, Inc. - All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""
+Netket python utilities.
+
+"""
+import json
+import numpy as np
+import matplotlib.pyplot as plt
+
+
+def message(type, msg):
+    if type == "Info":
+        print("# " + msg)
+
+    elif type == "Warning":
+        print("# WARNING: " + msg)
+
+    elif type == "Error":
+        print("# ERROR: " + msg)
+
+    elif type == "Debug":
+        print("# DEBUG: " + msg)
+
+    else:
+        raise ValueError("%s Message not supported" % type)
+
+
+def set_mand_pars(params, key, kwargs, def_value):
+    try:
+        params[key] = kwargs[key]
+    except KeyError:
+        params[key] = def_value
+
+
+def set_opt_pars(params, key, kwargs):
+    try:
+        params[key] = kwargs[key]
+    except KeyError:
+        pass
+
+
+def encode_complex(z):
+    if isinstance(z, complex):
+        return (z.real, z.imag)
+    else:
+        type_name = z.__class__.__name__
+        raise TypeError(
+            f"Object of type '{type_name}' is not JSON serializable")
+
+
+def get_obsv_from_json(outputfile):
```

Comment:
Can we also extract the variance and other properties available in the observable?

Reply:
We can extract the mean, sigma, and taucorr for any observable.